llama-index-core/llama_index/core/output_parsers/langchain.py

```diff
@@ -23,23 +23,26 @@ def __init__(
 
     def parse(self, output: str) -> Any:
         """Parse, validate, and correct errors programmatically."""
-        # TODO: this object may be stringified by our upstream llmpredictor,
-        # figure out better
-        # ways to "convert" the object to a proper string format.
+        # Convert output to string if it isn't already, handling various input types
+        if not isinstance(output, str):
+            try:
+                output = str(output).strip()
+            except Exception as e:
+                raise ValueError(f"Unable to convert output to string: {e}")
+        
         return self._output_parser.parse(output)
 
     def format(self, query: str) -> str:
         """Format a query with structured output formatting instructions."""
         format_instructions = self._output_parser.get_format_instructions()
 
-        # TODO: this is a temporary hack. if there's curly brackets in the format
-        # instructions (and query is a string template), we need to
-        # escape the curly brackets in the format instructions to preserve the
-        # overall template.
         query_tmpl_vars = {
             v for _, v, _, _ in Formatter().parse(query) if v is not None
         }
+        
         if len(query_tmpl_vars) > 0:
+            format_instructions = format_instructions.replace("{{", "{{{{")
+            format_instructions = format_instructions.replace("}}", "}}}}")
```

Comment:
Actually, we should just use the SafeFormatter here, no need to worry about brackets
https://github.com/run-llama/llama_index/blob/a657056cab608c494bdb4efa08fb60dc8f69699a/llama-index-core/llama_index/core/prompts/utils.py#L7

Reply:
Hi @logan-markewich, I updated using SafeFormatter, please take a look

Reply:
Did you push it? I still see `Formatter().parse(query)` and all the bracket replacements